### PR TITLE
Test conda 4.7.1

### PR DESCRIPTION
**Pull request by TomAugspurger**
Testing conda 4.7.x

### CODE REVIEW

**Comment by TomAugspurger** [comment]
The remaining failures are due to the `free` channel is no longer included. https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=12064&view=logs&jobId=d6a36677-9147-59f9-684d-001fc5c44c5e&taskId=9cd84fbe-5446-5224-e7d9-fb3364f72566&lineStart=305&lineEnd=306&colStart=1&colEnd=1 and https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=12064&view=logs&jobId=a69e7846-138e-5465-0656-921e8964615b&taskId=b5a8be37-ad64-5774-649a-2b5e4f73cc9a&lineStart=318&lineEnd=319&colStart=1&colEnd=1. Need to either bump those old package versions, remove those optional deps in those builds, or restore free.

**Comment by TomAugspurger** [comment]
To summarize, conda is removing the `free` channel by default, so some of our older packages won't be installed. Here are the packages requiring bumps:

library | old pin | minimum conda | release date
------- | ------- | ------------- | ------------
pytz | 2015.4 | 2017.2 | March 2017
beautifulsoup4 | 4.4.1 | 4.6.0 | May 2017
xlwt | 1.0.0  | 1.2.0 | January 2017
python-dateutil | 2.5.0 | 2.6.1 | July 2017
xlsxwriter | 0.7.7 | 0.9.8 | July 2017
openpyxl | 2.4.0 | 2.4.8 | May 2017

Our options

1. Bump our minimum supported versions to those
2. Move to pip for the ones that can't come from Conda anymore
3. Add back the free channel


I think 1 or 2 is fine. I have a slight preference for 1, but either works.

**Comment by WillAyd** [comment]
Releases are two years old at this point so I think option 1 is fine as well

**Comment by jreback** [comment]
I agree, ok with bumping

**Comment by TomAugspurger** [comment]
Hopefully this next one passes, and we'll be ready.

Reminder: that we don't want to merge this. I'll make another PR for any additional prep changes (right now, just bumping the min version for pymysql)

**Comment by TomAugspurger** [comment]
Bumping gcsfs to 0.2.2; That's needed for compatibility with recent versions of dask.

**Comment by TomAugspurger** [comment]
FYI I've reverted the Conda-canary install of 4.7.x

This has a few more changes (version bumps) we should merge before 4.7.3 is released, which is happening soonish.

**Comment by TomAugspurger** [comment]
Conda 4.7 is out now, so I think this is affecting PRs.

Merging later tonight if there are no objections.

File: pandas/compat/_optional.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "bs4": "4.6.0",
     "bottleneck": "1.2.1",
     "fastparquet": "0.2.1",
-    "gcsfs": "0.1.0",
+    "gcsfs": "0.2.2",
+    "lxml.etree": "3.8.0",
```
Comment (jreback) [inline pandas/compat/_optional.py]: prob should add pymysql here as well

**Pull request merged**